instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanStatusExtractor.java

```diff
@@ -40,7 +40,10 @@ public StatusCode extract(REQUEST request, @Nullable RESPONSE response, Throwabl
     if (response != null) {
       Integer statusCode = attributesExtractor.statusCode(request, response);
       if (statusCode != null) {
-        return HttpStatusConverter.statusFromHttpStatus(statusCode);
+        StatusCode statusCodeObj = HttpStatusConverter.statusFromHttpStatus(statusCode);
+        if (statusCodeObj == StatusCode.ERROR) {
+          return statusCodeObj;
+        }
```

Comment:
Please note this change in behavior.

What prompted this here is that on circular redirect, apache-httpasyncclient calls `Instrumenter.end()` with both a 302 response object AND and exception.

And our http client tests think that circular redirects should result in `StatusCode.ERROR`.